datafusion/core/src/dataframe.rs

```diff
@@ -1014,12 +1014,21 @@ impl DataFrame {
     /// ```
     pub fn with_column_renamed(
         self,
-        old_name: impl Into<Column>,
+        old_name: &str,
```

Comment:
Could be

```suggestion
        old_name: impl Into<String>,
```

To maintain backwards compatibility technically?